sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StatefulAggregate.scala

```diff
@@ -180,11 +181,41 @@ case class StateStoreSaveExec(
 
           // Update and output modified rows from the StateStore.
           case Some(Update) =>
+
+            // Note: Update mode can be done with or without a specified watermark
+            val optionalWatermarkAttribute =
+              keyExpressions.find(_.metadata.contains(EventTimeWatermark.delayKey))
+
+            val optionalPredicate = optionalWatermarkAttribute.map { watermarkAttribute =>
+              // If we are evicting based on a window, use the end of the window.  Otherwise just
```

Comment:
any reason why you wouldn't want to make this a method? Seems exactly the same as above under `append` mode